README.md

```diff
@@ -344,6 +345,58 @@ Variables are not restricted to only one assigned type. This is where union type
   // myRealRealAge = true // Not possible since myRealRealAge only accepts a number or a string.
 ```
 
+### Intersection
+
+Intersection types are similar to union types (and similar to the `extend` keyword pattern, which is explained further down), but arguably less common (yet, at least equally useful). Before we get into what an intersection type is, I must mention that in the [TypeScript official documentation about the intersection types](https://www.typescriptlang.org/docs/handbook/advanced-types.html), they are showcased first and *before* the union types, however in this guide I will mention them *after* the union types because they have a bit of a more abstract definition.
+
+To give you an idea of what an intersection type is, think of a mathematical intersection equal to [A ∩ B](https://upload.wikimedia.org/wikipedia/commons/thumb/9/99/Venn0001.svg/220px-Venn0001.svg.png), **but even then, that is not completely right in all cases**. In TypeScript, an intersection type is simply a mix (more commonly referred to as a ***mixin***) between two or more types.
```

Comment:
caution... it's actually not the same as the mathematical intersection, it's more of a merge i.e. the resulting type will have the properties of each type as long as intersection is possible because the intersected types are structurally equal/compatible

Reply:
Even though it's mentioned that it's not completely the same I agree that it might give the reader a wrong idea, I'll try to think of a different yet at least equally easier way to give the reader the right idea of intersection types without getting too much into complicated/relatively advanced topics!

Reply:
agree, makes sense, most of the time it's actually good to remove some complexity/quantity in order to provide useful information to the majority of people; those who want to drill down on details will do so anyway